### PR TITLE
!str #16937 Remove section, in favor of via and separate flow

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-docs-dev/rst/java/stream-error.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ elements that cause the division by zero are effectively dropped.
   Be aware that dropping elements may result in deadlocks in graphs with
   cycles, as explained in :ref:`graph-cycles-java`.
 
-The supervision strategy can also be defined for a section of flow operators.
+The supervision strategy can also be defined for all operators of a flow.
```
Comment (ktoso) [inline akka-docs-dev/rst/java/stream-error.rst]: "pendantic mode on": all stages :)

File: akka-docs-dev/rst/scala/code/docs/stream/IntegrationDocSpec.scala [hunk 2]
```diff
@@ -177,9 +178,9 @@ class IntegrationDocSpec extends AkkaSpec(IntegrationDocSpec.config) {
     import Supervision.resumingDecider
 
     val emailAddresses: Source[String, Unit] =
-      authors.section(supervisionStrategy(resumingDecider)) {
-        _.mapAsync(author => addressSystem.lookupEmail(author.handle))
-      }
+      authors.via(
+        Flow[Author].mapAsync(author => addressSystem.lookupEmail(author.handle))
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/stream/IntegrationDocSpec.scala]: Unrelated: Was wondering if we could make `.via(_.mapAsync{})` constructs work...

**Comment by ktoso** [comment]
LGTM

File: akka-docs-dev/rst/scala/code/docs/stream/IntegrationDocSpec.scala (hunk 2 above)
Comment (rkuhn) [inline akka-docs-dev/rst/scala/code/docs/stream/IntegrationDocSpec.scala]: no, we cannot: function literal input type inference only works for non-overloaded methods

**Comment by rkuhn** [comment]
LGTM: nice Java workout :-)

**Comment by patriknw** [comment]
rebased, one more time

**Comment by patriknw** [comment]
Refs #16937

**Pull request merged**